lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Fonts Library class.
+ *
+ * This file contains the Fonts Library class definition.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts Library
+ * @since      X.X.X
+ */
+
+if ( class_exists( 'WP_Fonts_Library' ) ) {
+	return;
+}
+
+/**
+ * Fonts Library class.
+ */
+class WP_Fonts_Library {
+
+	const ALLOWED_FONT_MIME_TYPES = array(
+		'otf'   => 'font/otf',
+		'ttf'   => 'font/ttf',
+		'woff'  => 'font/woff',
+		'woff2' => 'font/woff2',
+	);
+
+	/**
+	 * Get the upload directory for fonts.
+	 *
+	 * @return string Path of the upload directory for fonts.
+	 */
+	public static function get_fonts_dir() {
+		return wp_upload_dir()['basedir'] . '/fonts';
+	}
+
+	/**
+	 * Sets the upload directory for fonts.
+	 *
+	 * @param array $defaults Default upload directory.
```

Comment:
As the description makes it sound like a string, let's [document the contents of the array](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-1-parameters-that-are-arrays) so it's clear what to pass.